.github/workflows/cpp_windows.yml

```diff
@@ -63,10 +64,11 @@ jobs:
       ARROW_WITH_ZLIB: ON
       ARROW_WITH_ZSTD: ON
       BOOST_SOURCE: BUNDLED
-      CMAKE_CXX_STANDARD: "17"
```

Comment:
Gtest can't be used with C++17 so removing this line. I see `CMAKE_CXX_STANDARD` isn't directly specified in `cpp.yml` file as well